docs/docsite/rst/reference_appendices/module_utils.rst

```diff
@@ -25,3 +25,50 @@ To use this functionality, include ``import ansible.module_utils.basic`` in your
 
 .. automodule:: ansible.module_utils.basic
    :members:
+
+
+Argument Spec
+---------------------
+
+Classes and functions for validating parameters against an argument spec.
+
+ArgumentSpecValidator
+=====================
+
+.. autoclass:: ansible.module_utils.common.arg_spec.ArgumentSpecValidator
+   :members:
+
+ValidationResult
+================
+
+.. autoclass:: ansible.module_utils.common.arg_spec.ValidationResult
+   :members:
+   :member-order: bysource
+   :private-members: _no_log_values  # This only works in sphinx >= 3.2. Otherwise it shows all private members with doc strings.
```

Comment:
Since we aren't at 3.2 or above, is this going to cause display problems for 2.11 docs?

Reply:
No, it will display just fine with Sphinx 2.1.2, but only because no other private things have docstrings.

This comment is mainly to document the behavior that took me quite some time to figure out what was going on. `:private-members:` was rendering the docs for _all_ private methods/attributes even if they weren't in the list. It's because it wasn't until Sphinx 3.2 that `:private-members:` honored the values passed in.